lib/helpers/url-utils.js

```diff
@@ -54,7 +54,7 @@ function appendQueryObject(url, queryObject) {
     if (value === undefined) {
       return;
     }
-    if (Array.isArray(value)) {
+    if (Array.isArray(value) && key !== 'setfilter') {
```

Comment:
📝I think we need to keep this to make sure `setfilter` retains its array shape

Reply:
Sorry for making things extra confusing, but I think it is a bit cleaner to stringify in the static service and leave the special case out of `url-utils`. That way we can keep `url-utils` generalized and protect ourselves from dealing with more service-specific edge cases in the future. 

What do you think?